internal/server/v1/event/event.go

```diff
@@ -27,6 +30,40 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+var noDataPlaces = map[string]struct{}{
+	"Earth":        {},
+	"africa":       {},
+	"antarctica":   {},
+	"asia":         {},
+	"europe":       {},
+	"northamerica": {},
+	"oceania":      {},
+	"southamerica": {},
+}
+
+type filterSpec struct {
+	prop       string
+	unit       string
+	lowerLimit float64
+	upperLimit float64
+}
+
+func keepEvent(event *pb.EventCollection_Event, spec filterSpec) bool {
+	for prop, vals := range event.GetPropVals() {
+		if prop == spec.prop {
+			valStr := strings.TrimPrefix(vals.Vals[0], spec.unit)
+			v, err := strconv.ParseFloat(valStr, 64)
+			if err != nil {
+				// Can not convert the value, keep.
+				return true
```

Comment:
this is fine for now if the idea here is to reduce data size. can we add a note of that principle with the request or here in the function?

Reply:
Added function comment.